File: py/utils/netcat/netcat.py
#!/usr/bin/python3

import socket
import sys


if __name__ == "__main__" :
	s = socket.socket()


	host = sys.argv[1]
	port = sys.argv[2]

	s.connect(  (host, int(port) )  )

	while True :
		string = input()
		s.send(string.encode())
		data = s.recv(1024)
		print("'"+ data.decode() +"'") 

File: pl/5/utils/network/netcat/netcat.pl
#!/usr/bin/env perl
use strict; 
use Socket;
use feature 'say' ;

# Arguments getting.
my $host = $ARGV[0] ;
my $port = $ARGV[1] ;
my $bin_ip = inet_aton($host) ;

socket(SOCK, PF_INET, SOCK_STREAM, getprotobyname('tcp'));
my $paddr = sockaddr_in($port, $bin_ip) ;

connect(SOCK, $paddr) or die "connect:$!\n" ;
my $data;
do{
		my $in = <STDIN> ;
		send SOCK, $in, 0 or die "send: $!" ;

		print $data while ( $data = <SOCK>) ;
}while($data);

say STDERR "$0: exiting program." ;
exit;
